WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderTagHeaderView.java

```diff
@@ -40,64 +47,81 @@ public ReaderTagHeaderView(Context context, AttributeSet attrs, int defStyleAttr
 
     private void initView(Context context) {
         View view = inflate(context, R.layout.reader_tag_header_view, this);
-        mFollowButton = (ReaderFollowButton) view.findViewById(R.id.follow_button);
+        mImageView = (WPNetworkImageView) view.findViewById(R.id.image_tag_header);
+        mTxtAttribution = (TextView) view.findViewById(R.id.text_attribution);
     }
 
     public void setCurrentTag(final ReaderTag tag) {
         if (tag == null) return;
 
+        if (!ReaderTag.isSameTag(tag, mCurrentTag)) {
+            mTxtAttribution.setText(null);
+            mImageView.resetImage();
+        }
+
         mCurrentTag = tag;
 
         TextView txtTagName = (TextView) findViewById(R.id.text_tag);
-        txtTagName.setText(ReaderUtils.makeHashTag(tag.getTagSlug()));
-
-        if (ReaderUtils.isLoggedOutReader()) {
-            mFollowButton.setVisibility(View.GONE);
-        } else {
-            mFollowButton.setVisibility(View.VISIBLE);
-            mFollowButton.setIsFollowed(ReaderTagTable.isFollowedTagName(tag.getTagSlug()));
-            mFollowButton.setOnClickListener(new OnClickListener() {
-                @Override
-                public void onClick(View v) {
-                    toggleFollowStatus();
-                }
-            });
-        }
+        txtTagName.setText(tag.getLabel());
+
+        getImageAndAttribution();
     }
 
-    private void toggleFollowStatus() {
-        if (mCurrentTag == null || !NetworkUtils.checkConnection(getContext())) {
-            return;
-        }
+    private void getImageAndAttribution() {
```

Comment:
I wasn't able to get rid of the extra `onBindViewHolder` call - I'm not sure if that's just the way RecyclerView works due to cell resizing, etc. At any rate, I added a cache of header info so the network request doesn't need to be made more than once.